doc/kernel/services/sys_irq.rst

```diff
@@ -0,0 +1,58 @@
+.. _sys_irq:
+
+SYS IRQ (SYStem Interrupt ReQuest) WIP
+######################################
+
+The SYS IRQ kernel service utilizes the devicetree and device driver model to
+provide interrupt handling scalably, portably and efficiently.
+
+Design
+******
+
+The design provides clear separation between interrupt controllers, interrupt
+generating devices, and interrupt request management.
+
+IRQ (Interrupt ReQuests)
+========================
+
+An IRQ is a hardware event which, when triggered, invokes a handler in
+software.
+
+INTC (INTerrupt Controller)
+===========================
+
+INTCs invoke SYS IRQ INTL (INTerrupt Line) handlers when an interrupt is detected
+on the corresponding INTL.
+
+It is the responsibility of the INTC device driver, SoC and Arch, to do any and
+all preparation required to invoke the SYS IRQ INTL handlers from a execution
+context.
+
+Any and all configuration of INTLs is performed by SYS IRQ through the INTC device
+driver API.
+
+INTD (INTerrupt generating Device)
```

Comment:
I think the abbreviation and the full name should be separated. Simply put, the whole process should be capitalized according to the first letter of the sentence, without deliberately showing the abbreviation.

For example
```
INTD (Interrupt generating device)
```

(Applicable to all changes.)